Provides workflow for building from Debian-maintained unofficial patches

conorsch opened the pull request:
Adds a new Molecule scenario that boasts the following features:

* Uses the Debian-maintained "unofficial" grsec patches
* Does *not* require subscription access
* Runs in containers for a light workload (#6)
* Includes a bundled config file for fully unattended builds

Try it out with:

```
molecule test -s unofficial
```

Note that'll take a _long time_—several hours or so—since it's going to run a full build. The improvements in #7 and #8 are strongly recommended for reliability, but aren't strictly necessary. 

Regarding diff review, it's mostly Molecule boilerplate and some task logic that was moved around. Tried hard to reuse as much existing task logic as possible. Now the `tasks/main.yml` file splits the flow based on patch type, which seems reasonable. See commit messages for detailed explanations of the changes.

Closes #2.

msheiny commented:
Marge conflicts reported on this one ^^

conorsch commented:
Thanks, rebase incoming.

conorsch commented:
Rebased!

msheiny commented on molecule/unofficial/tests/test_default.py:
```diff
@@ -0,0 +1,14 @@
+import os
```
You should probably strip out this default test file

msheiny commented on molecule/unofficial/molecule.yml:
```diff
@@ -0,0 +1,20 @@
+---
```
The run list for `test` by default includes:

```

└── unofficial
    ├── destroy
    ├── dependency
    ├── syntax
    ├── create
    ├── prepare
    ├── converge
    ├── idempotence
    ├── lint
    ├── side_effect
    ├── verify
    └── destroy
```

I would remove `idempotence`, `prepare`, and  `side_effect`

conorsch replied:
Good call on removing the unnecessary steps, will do that!

msheiny commented:
I have larger concerns related to use of ansible for this task (building a grsec kernel). I suspect it may be easier to maintain + faster if you folded some of the build logic into docker and where its too annoying, fold together a bunch of the ansible tasks into a `script` task. I do like keeping the role in ansible though because it allows easy build on remote machines. I think a lot of the build doesnt really call for idempotency (lots of `shell` and `command` tasks) and ansible just makes the build logic more complex vs. a bash script to do the heavy lifting. Not asking for you to look at that here -- maybe a separate ticket - im happy to help with that process if you dont object.

conorsch commented:
Addressed your concerns about the excessive testing flow. Commented out the problematic linting and verify steps. Left the default testinfra test file—now unused—so we can modify it in place once we have a better testing story.
